src/braket/simulator_v2/base_simulator_v2.py

```diff
@@ -153,20 +152,22 @@ def run_openqasm(
         if shots:
             r.resultTypes = results
         else:
-            r = _analytic_result_value_to_ndarray(r)
+            r = _result_value_to_ndarray(r)
         return r
 
 
-def _analytic_result_value_to_ndarray(
+def _result_value_to_ndarray(
     task_result: GateModelTaskResult,
 ) -> GateModelTaskResult:
     """Convert any StateVector or DensityMatrix result values from raw Python lists to the expected
     np.ndarray. This must be done because the wrapper Julia simulator results Python lists to comply
     with the pydantic specification for ResultTypeValues.
     """
     for result_ind, result_type in enumerate(task_result.resultTypes):
-        if isinstance(result_type.type, StateVector) or isinstance(
-            result_type.type, DensityMatrix
+        if (
+            isinstance(result_type.type, StateVector)
+            or isinstance(result_type.type, DensityMatrix)
+            or isinstance(result_type.type, Probability)
```

Comment:
Can you add unit tests for the different result types we expect to treat this way?